### PR TITLE
RHOBS-957: Add option to override CLI args

### PR DESCRIPTION
**Pull request by jacobbaungard**
With this commit it is possible to override CLI args for various Thanos components. As an example, here is how one would override the receivers CLI args:

```
thanos:
  receivers:
    containers:
      - name: thanos-receive
	args:
	- receive
	- --log.level=debug
	- --log.format=logfmt
	- --grpc-address=0.0.0.0:10901
	- --http-address=0.0.0.0:10902
	- --remote-write.address=0.0.0.0:19291
	- --receive.replication-factor=1
	- --tsdb.path=/var/thanos/receive
	- --tsdb.retention=4d
	- --label=replica="$(NAME)"
	- --label=receive="true"
	- --objstore.config=$(OBJSTORE_CONFIG)
	- --receive.local-endpoint=$(NAME).observatorium-xyz-thanos-receive-default.$(NAMESPACE).svc.cluster.local:10901
	- --receive.hashrings-file=/var/lib/thanos-receive/hashrings.json
	- --receive.default-tenant-id="default-tenant-test"
```

### CODE REVIEW

**Review by moadz — APPROVED** [review]
LGTM

**Review by saswatamcode — APPROVED** [review]
/lgtm

File: jsonnet/obs-operator.jsonnet [hunk 1]
```diff
@@ -415,6 +424,77 @@ local operatorObs = obs {
...
+        std.startsWith(v.metadata.name, cr.metadata.name + '-thanos-store-memcached') &&
+        std.objectHas(cr.spec.thanos.store.cache, 'containers')) then {
+        template+: {
+          spec+: {
+            containers: override_containers(super.containers, cr.spec.thanos.store.containers)
+          },
+        },
+      } else {}
+    ) + (
+      // Will this also apply to QFE??
```
Comment (saswatamcode) [inline jsonnet/obs-operator.jsonnet:458]: Possibly, as we match with "thanos-query"?

Comment (jacobbaungard) [reply jsonnet/obs-operator.jsonnet:458]: oh opps, I forgot about that...

**Pull request merged**